tests/test_discriminated_union.py

```diff
@@ -13,22 +13,49 @@
 from pydantic.errors import PydanticUserError
 
 
-def test_discriminated_union_only_union():
+def test_discriminated_union_type():
     with pytest.raises(
-        TypeError, match='`discriminator` can only be used with `Union` type with more than one variant'
+        TypeError, match="'str' is not a valid discriminated union variant; should be a `BaseModel` or `dataclass`"
     ):
 
         class Model(BaseModel):
             x: str = Field(..., discriminator='qwe')
 
 
+@pytest.mark.parametrize('union', [True, False])
+def test_discriminated_single_variant(union):
+    class InnerModel(BaseModel):
+        qwe: Literal['qwe']
+        y: int
+
+    class Model(BaseModel):
+        if union:
+            x: Union[InnerModel] = Field(..., discriminator='qwe')
+        else:
+            x: InnerModel = Field(..., discriminator='qwe')
```

Comment:
`Union[InnerModel]` just evaluates to `InnerModel` and I think this is unlikely to change, so there may be no need to test the two branches?

ref https://docs.python.org/3/library/typing.html#typing.Union